lib/desktop/quicksettings/quick_settings.dart

```diff
@@ -341,83 +342,46 @@ class QuickSettingsState extends State<QuickSettings> {
-                      Pangolin.settingsBox.put("language", "bs_BA");
-                      break;
-                    case "bs_BA":
-                      Pangolin.setLocale(context, Locale("hr", "HR"));
-                      Pangolin.settingsBox.put("language", "hr_HR");
-                      break;
-                    case "hr_HR":
-                      Pangolin.setLocale(context, Locale("nl", "NL"));
-                      Pangolin.settingsBox.put("language", "nl_NL");
-                      break;
-                    case "nl_NL":
-                      Pangolin.setLocale(context, Locale("fr", "FR"));
-                      Pangolin.settingsBox.put("language", "fr_FR");
-                      break;
-                    case "fr_FR":
-                      Pangolin.setLocale(context, Locale("de", "DE"));
-                      Pangolin.settingsBox.put("language", "de_DE");
-                      break;
-                    case "de_DE":
-                      Pangolin.setLocale(context, Locale("id", "ID"));
-                      Pangolin.settingsBox.put("language", "id_ID");
-                      break;
-                    case "id_ID":
-                      Pangolin.setLocale(context, Locale("pl", "PL"));
-                      Pangolin.settingsBox.put("language", "pl_PL");
-                      break;
-                    case "pl_PL":
-                      Pangolin.setLocale(context, Locale("pt", "BR"));
-                      Pangolin.settingsBox.put("language", "pt_BR");
-                      break;
-                    case "pt_BR":
-                      Pangolin.setLocale(context, Locale("ru", "RU"));
-                      Pangolin.settingsBox.put("language", "ru_RU");
-                      break;
-                    case "ru_RU":
-                      Pangolin.setLocale(context, Locale("sv", "SE"));
-                      Pangolin.settingsBox.put("language", "sv_SE");
-                      break;
-                    case "sv_SE":
-                      Pangolin.setLocale(context, Locale("uk", "UA"));
-                      Pangolin.settingsBox.put("language", "uk_UA");
-                      break;
-                    case "uk_UA":
-                    default:
-                      Pangolin.setLocale(context, Locale("en", "US"));
-                      Pangolin.settingsBox.put("language", "en_US");
-                      break;
+                buildTile(Icons.airplanemode_inactive,
+                    LocaleStrings.pangolin.qsAirplanemode, changeColor),
+                buildTile(Icons.invert_colors_off,
+                    LocaleStrings.pangolin.qsInvertcolors, changeColor),
+                buildTile(
+                    Icons.language, LocaleStrings.pangolin.qsChangelanguage,
+                    () {
+                  int newLocale = 0;
+                  // final currentLanguage = context.locale.countryCode;
+                  for (var locale in Locales.supported) {
```

Comment:
seems far too complicated still, just do an indexOf in the supported list using the current locale, get the index, and just set new locale to index + 1

Reply:
so, I replace it with a if/else, for case get the range of supported locales. 🤔

Reply:
int index = Locales.supported.indexOf(context.locale);
if(index >= Locales.supported.length)
  index = -1;
context.locale = Locales.supported[index + 1];

something like this

Reply:
I did as 
```
if (index+1 < supported.length)
context.locale = Locales.supported[index + 1];
```
Its a pretty closes 😅